code/game/objects/items/mountable_frames/posters/poster.dm

```diff
@@ -117,6 +117,52 @@ obj/structure/sign/poster/attackby(obj/item/weapon/W as obj, mob/user as mob)
 		H.apply_damage(rand(5,7), BRUTE, pick(LIMB_RIGHT_LEG, LIMB_LEFT_LEG, LIMB_RIGHT_FOOT, LIMB_LEFT_FOOT))
 
 
+
+/obj/item/mounted/poster/explosive
+	name = "Commendation Poster"
+	desc = "The poster comes with its own automatic adhesive mechanism, for easy pinning to any vertical surface. Remove at personal risk."
+	icon = 'icons/obj/posters.dmi'
+	icon_state = "rolled_poster"
+	w_type=RECYK_MISC
+
+/obj/item/mounted/poster/explosive/do_build(turf/on_wall, mob/user)
+	to_chat(user, "<span class='notice'>You start placing the poster on the wall...</span>")
+	var/obj/structure/sign/poster/explosive/D = new(src)
```

Comment:
You're creating this object inside the src, then moving it onto the wall after that. You could just do `var/obj/structure/sign/poster/explosive/D = new(on_wall)` and remove the `forceMove()`